ansible_base/authentication/utils/authentication.py

```diff
@@ -0,0 +1,120 @@
+import logging
+from typing import Optional, Tuple
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AbstractUser
+from social_core.pipeline.user import get_username
+
+from ansible_base.authentication.models import Authenticator, AuthenticatorUser
+from ansible_base.authentication.social_auth import AuthenticatorStorage, AuthenticatorStrategy
+
+logger = logging.getLogger('ansible_base.authentication.utils.authentication')
+
+
+def get_local_username(user_details: dict) -> str:
+    """
+    Converts the username provided by the backend to one that doesn't conflict with users
+    from other auth backends.
+    """
+
+    class FakeBackend:
```

Comment:
Is there a reason to use nested class for this function?

Reply:
This code is actually not new but was split out from ansible_base/authentication/utils/claims.py. I don't know why it was initially implemented this way but we can move it out since its really nothing dynamic.=